dpctl/_sycl_queue.pyx

```diff
@@ -1524,3 +1539,41 @@ cdef api SyclQueue SyclQueue_Make(DPCTLSyclQueueRef QRef):
     """
     cdef DPCTLSyclQueueRef copied_QRef = DPCTLQueue_Copy(QRef)
     return SyclQueue._create(copied_QRef)
+
+cdef class _WorkGroupMemory:
+    def __dealloc__(self):
+        if(self._mem_ref):
+            DPCTLWorkGroupMemory_Delete(self._mem_ref)
+
+cdef class WorkGroupMemory:
+    """
+    WorkGroupMemory(nbytes)
+    Python class representing the ``work_group_memory`` class from the
+    Workgroup Memory oneAPI SYCL extension for low-overhead allocation of local
+    memory shared by the workitems in a workgroup.
+
+    This is based on a DPC++ SYCL extension and only available in newer
+    versions. Use ``is_available()`` to check availability in your build.
+
+    Args:
+        nbytes (int)
+            number of bytes to allocate in local memory.
+            Expected to be positive.
+    """
+    def __cinit__(self, Py_ssize_t nbytes):
```

Comment:
Do you see a reason to provide another constructor that takes a given type and a given number of elements? The extension includes `work_group_memory(size_t num, handler& cgh)`

could be as simple as a a call which wraps conversion from `num` and `nelems` to `nbytes`, and could be handled similarly to `SyclQueue` (`*args` and check the length, when it's two, use `num` and `dtype` overload, when it's one, `nbytes`)

only question is if there's much of a need or use for it

Reply:
I've implemented such a constructor now. It's now possible to construct the `WorkGroupMemory` with either a number of bytes directly or a data type (using `numpy` type names) and an element count.

For our purpose, we wouldn't use that second constructor, but I think the Python test illustrates that it's a nice addition.